client/src/pages/reports/Form.tsx

```diff
@@ -1605,6 +1605,18 @@ const ReportForm = ({
 
   function save(values, sendEmail) {
     const report = Report.filterClientSideFields(new Report(values))
+    // Check engagement date in range of engagement event if present
+    if (
+      report.event &&
+      !(
+        report.engagementDate >= report.event.startDate &&
+        report.engagementDate <= report.event.endDate
+      )
```

Comment:
IMHO the condition is more readable when inverted:
```javascript
      (
        report.engagementDate < report.event.startDate ||
        report.engagementDate > report.event.endDate
      )
```
but also be aware that `report.engagementDate` can still be empty here.